tests/unit/YourCompany.spec.ts

```diff
@@ -86,9 +90,102 @@ describe('YourCompany in a BEN correction', () => {
   })
 })
 
+const tests = [
+  {
+    entityType: 'BEN',
+    resourceModel: BenCorrectionResource,
+    isNumberedCompany: true,
+    expectedOptions: ['correct-new-nr']
+  },
+  {
+    entityType: 'BEN',
+    resourceModel: BenCorrectionResource,
+    isNumberedCompany: false,
+    expectedOptions: ['correct-new-nr', 'correct-name-to-number', 'correct-name']
+  },
+  {
+    entityType: 'CC',
+    resourceModel: CccCorrectionResource,
+    isNumberedCompany: true,
+    expectedOptions: ['correct-new-nr']
+  },
+  {
+    entityType: 'CC',
+    resourceModel: CccCorrectionResource,
+    isNumberedCompany: false,
+    expectedOptions: ['correct-new-nr', 'correct-name-to-number', 'correct-name']
+  },
+  {
+    entityType: 'GP',
+    resourceModel: GpCorrectionResource,
+    isNumberedCompany: false,
+    expectedOptions: ['correct-new-nr']
+  },
+  {
+    entityType: 'BC',
+    resourceModel: BcCorrectionResource,
+    isNumberedCompany: true,
+    expectedOptions: ['correct-new-nr']
+  },
+  {
+    entityType: 'BC',
+    resourceModel: BcCorrectionResource,
+    isNumberedCompany: false,
+    expectedOptions: ['correct-new-nr', 'correct-name-to-number', 'correct-name']
+  },
+  {
+    entityType: 'SP',
+    resourceModel: SpCorrectionResource,
+    isNumberedCompany: false,
+    expectedOptions: ['correct-new-nr']
+  },
+  {
+    entityType: 'ULC',
+    resourceModel: UlcCorrectionResource,
+    isNumberedCompany: true,
+    expectedOptions: ['correct-new-nr']
+  },
+  {
+    entityType: 'ULC',
+    resourceModel: UlcCorrectionResource,
+    isNumberedCompany: false,
+    expectedOptions: ['correct-new-nr', 'correct-name-to-number', 'correct-name']
+  }
+]
+
+for (const test of tests) {
+  const type = test.isNumberedCompany ? 'numbered' : 'named'
+
+  describe('Name Change Options in a correction', () => {
```

Comment:
I only tested this for corrections. Do you think it's OK?

Reply:
Do we have any ticket where we can can add/list files to add unit tests in future? Just not miss them and add later.

Reply:
It would be nice to have an epic which we could add to for tracking this sort of hidden work that needs to be done.

Reply:
No generic ticket. We create specific tickets as needed.

Since a couple of you think this testing is a good idea, I'll just add it in now.

Reply:
We don't have such an epic yet, but we do have the `Tech Debt` label on tickets.

Reply:
Done.